lib/Dialect/Moore/MooreTypes.cpp

```diff
@@ -20,49 +20,13 @@ using namespace circt;
 using namespace circt::moore;
 using mlir::AsmParser;
 using mlir::AsmPrinter;
-using mlir::LocationAttr;
-using mlir::OptionalParseResult;
-using mlir::StringSwitch;
-using mlir::TypeStorage;
-using mlir::TypeStorageAllocator;
 
-//===----------------------------------------------------------------------===//
-// Generated logic
-//===----------------------------------------------------------------------===//
-
-#define GET_TYPEDEF_CLASSES
-#include "circt/Dialect/Moore/MooreTypes.cpp.inc"
-
-void MooreDialect::registerTypes() {
-  addTypes<PackedStructType, UnpackedStructType>();
-
-  addTypes<
-#define GET_TYPEDEF_LIST
-#include "circt/Dialect/Moore/MooreTypes.cpp.inc"
-      >();
-}
-
-//===----------------------------------------------------------------------===//
-// Utilities
-//===----------------------------------------------------------------------===//
-
-StringRef moore::getKeywordFromSign(const Sign &sign) {
-  switch (sign) {
-  case Sign::Unsigned:
-    return "unsigned";
-  case Sign::Signed:
-    return "signed";
-  }
-  llvm_unreachable("all signs should be handled");
-}
-
-struct Subset {
-  enum { None, Unpacked, Packed } implied = None;
-  bool allowUnpacked = true;
-};
+static LogicalResult parseMembers(AsmParser &parser,
+                                  SmallVector<StructMember> &members);
+static void printMembers(AsmPrinter &printer, ArrayRef<StructMember> members);
 
-static ParseResult parseMooreType(AsmParser &parser, Subset subset, Type &type);
-static void printMooreType(Type type, AsmPrinter &printer, Subset subset);
+static ParseResult parseMooreType(AsmParser &parser, Type &type);
+static void printMooreType(Type type, AsmPrinter &printer);
```

Comment:
Could I consider deleting `Subset subset` here is different from #7090 which specifies `Subset::None` is due to manually skipping the estimation of packed and unpacked, and then doesn't print `packed<>` and `unpacked<>`, right?    However, in this PR, in order to move types into ODS and streamline the parsing/printing, the distinction of `Subset` is unnecessary.

Reply:
Yes exactly! Now that the types indicate in their name whether they are packed or not (`array` vs. `uarray`), that whole complicated `Subset`-based parsing/printing is no longer needed :smiley: